netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -607,7 +644,27 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
           logger.log(Level.WARNING, String.format("Received unexpected ping ack. "
               + "Expecting %d, got %d", p.payload(), ackPayload));
         }
-      } else {
+      }
+      else if (data.readLong() == 1234){     
+        pingreturn++;
+        int target = dataSinceLastPing * 2;
+        if(target > 8000000){
+          target = 0;
+        }
+        pinging = false;
+        logger.log(Level.FINER, "OBDP: " + dataSinceLastPing);
```

Comment:
You probably want to either use a format string for this or wrap it in a isLoggable() call.    Java is going to do this concatenation on every call, and then discard it most of the time since the log level is low.